llvm/lib/Target/PowerPC/AsmParser/PPCAsmParser.cpp

```diff
@@ -99,10 +100,14 @@ struct PPCOperand;
 class PPCAsmParser : public MCTargetAsmParser {
   bool IsPPC64;
 
+  bool UsesMemOp;
```

Comment:
I don't think this statefulness is good. Instead of trying to record (in an instance variable of the parser) that a memory operand was encountered, perhaps it is possible to instead mark the "operand" that is created to represent the base register of the memory operand.